### ISSUE TITLE
QLC driver has pointless version checks

### ISSUE THREAD

**Issue by andy-js** [gdamore/illumos-core#165]
It's in the gate so checks like this are probably unnecessary: https://github.com/gdamore/illumos-core/blob/core/usr/src/uts/common/io/fibre-channel/fca/qlc/ql_api.c#L610

**Comment by gdamore** [comment]
Agreed.  There are probably other checks through the code that are intended to “configure” operation appropriately per release level.  We should fix these; simplifying the driver would be good.  It might also be a good time to review the driver for other “legacy” things, for example, updating the driver to use SCSAv3 interfaces for DMA, etc.

	- Garrett
